pkg/rtc/participant.go

```diff
@@ -499,9 +501,16 @@ func (p *ParticipantImpl) Close(sendLeave bool) error {
 		})
 	}
 
-	if p.dataTrack != nil {
-		p.dataTrack.Close()
+	var dt *DataTrack
+	p.lock.Lock()
+	dt = p.dataTrack
+	p.dataTrack = nil
```

Comment:
previously this was not set to `nil` after closing. And there is no `AddOnClose` registered on the data track also. So, set it to `nil` here.